sdk/include/opentelemetry/sdk/metrics/observer_result.h

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifndef ENABLE_METRICS_PREVIEW
+#  include "opentelemetry/common/key_value_iterable.h"
+#  include "opentelemetry/metrics/observer_result.h"
+#  include "opentelemetry/sdk/metrics/state/attributes_hashmap.h"
+#  include "opentelemetry/sdk/metrics/view/attributes_processor.h"
+
+#  include <map>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+template <class T>
+class ObserverResult final : public opentelemetry::metrics::ObserverResult<T>
+{
+public:
+  ObserverResult(const AttributesProcessor *attributes_processor)
+      : attributes_processor_(attributes_processor)
+  {}
+
+  virtual void Observe(T value) noexcept override { data_.insert({{}, value}); }
+
+  virtual void Observer(T value,
```

Comment:
typo -> `Observe`

Reply:
Nice catch. This typo got cascaded from `api::ObserverResult` too and also got added in tests. Have fixed in all the places now.